impeller/renderer/backend/vulkan/device_buffer_vk.h

```diff
@@ -6,20 +6,47 @@
 
 #include "flutter/fml/macros.h"
 #include "impeller/base/backend_cast.h"
+#include "impeller/renderer/backend/vulkan/debug_context_vk.h"
 #include "impeller/renderer/device_buffer.h"
 
 namespace impeller {
 
+class DeviceBufferAllocationVK {
+ public:
+  DeviceBufferAllocationVK(const VmaAllocator& allocator,
+                           VkBuffer buffer,
+                           VmaAllocation allocation,
+                           VmaAllocationInfo allocation_info);
+
+  ~DeviceBufferAllocationVK();
+
+  vk::Buffer GetBufferHandle() const;
+
+  void* GetMappedPtr() const;
+
+ private:
+  const VmaAllocator& allocator_;
+  vk::Buffer buffer_;
+  VmaAllocation allocation_;
+  VmaAllocationInfo allocation_info_;
```

Comment:
Here and elsewhere, `FML_DISALLOW_COPY_AND_ASSIGN`

Reply:
Done!